yb_migrate/cmd/importDataFileCommand.go

```diff
@@ -0,0 +1,264 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/yugabyte/yb-db-migration/yb_migrate/src/datafile"
+	"github.com/yugabyte/yb-db-migration/yb_migrate/src/tgtdb"
+	"github.com/yugabyte/yb-db-migration/yb_migrate/src/utils"
+)
+
+var (
+	fileType            string
+	delimiter           string
+	dataDir             string
+	fileTableMapping    string
+	hasHeader           bool
+	tableNameVsFilePath = make(map[string]string)
+	supportedFileTypes  = []string{datafile.CSV, datafile.SQL}
+	importDataFileMode  bool
+)
+
+var importDataFileCmd = &cobra.Command{
+	Use:   "file",
+	Short: "This command imports data from given files into YugabyteDB database",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		importDataFileMode = true
+		checkFileType()
+		checkDataDirFlag()
+		parseFileTableMapping()
+		prepareForImportDataCmd()
+		importDataFile()
+	},
+}
+
+func importDataFile() {
+	utils.PrintAndLog("import of data in %q database started", target.DBName)
+	err := target.DB().Connect()
+	if err != nil {
+		utils.ErrExit("Failed to connect to the target DB: %s", err)
+	}
+	fmt.Printf("Target YugabyteDB version: %s\n", target.DB().GetVersion())
+
+	sourceDBType = ORACLE // dummy value
+
+	dataFileDescriptor = datafile.OpenDescriptor(exportDir)
+	targets := getYBServers()
+
+	var parallelism = parallelImportJobs
+	if parallelism == -1 {
+		parallelism = len(targets)
+	}
+	log.Infof("parallelism=%v", parallelism)
+
+	if loadBalancerUsed {
+		clone := target.Clone()
+		clone.Uri = getTargetConnectionUri(clone)
+		targets = []*tgtdb.Target{clone}
+	}
+	if target.VerboseMode {
+		fmt.Printf("Number of parallel imports jobs at a time: %d\n", parallelism)
+	}
+
+	if parallelism > SPLIT_FILE_CHANNEL_SIZE {
+		splitFileChannelSize = parallelism + 1
+	}
+	splitFilesChannel := make(chan *SplitFileImportTask, splitFileChannelSize)
+	targetServerChannel := make(chan *tgtdb.Target, 1)
+
+	go roundRobinTargets(targets, targetServerChannel)
+	generateSmallerSplits(splitFilesChannel)
+	go doImport(splitFilesChannel, parallelism, targetServerChannel)
+	checkForDone()
+
+	time.Sleep(time.Second * 2)
+	fmt.Printf("\nexiting...\n")
```

Comment:
Can't we directly call the `importData()`?

That will require some refactoring to not to create indexes.

Reply:
will try doing that as part of the next commit.

Reply:
Yes. This is required. We're duplicating a lot of code.

You can do a simple refactoring so that `importData` accepts `sourceDBType` as argument.

Reply:
ok, i can rather set the `sourceDBType` var as a part of `Run` section of the command so it will not require a change in the function signature.

Reply:
done